js/src/App.vue

```diff
@@ -21,17 +21,35 @@ export default {
       tournaments: [],
       // The main websocket object
       ws: null,
+      reconnections: 0,
       user: new User(),
     }
   },
   methods: {
+    reconnect: function () {
+      this.$delete("ws")
+
+      if (this.reconnections < 30) {
+        console.warn(`closed uncleanly, reconnecting (try number ${this.reconnections})`)
+        setTimeout(() => {
+          this.reconnections = this.reconnections + 1
+          this.connect()
+        }, 500)
```

Comment:
I'm not fully following what breaks the reconnection loop. Could you fill me in?